backend/src/main/java/com/google/models/Alert.java

```diff
@@ -54,6 +58,10 @@ public StatusType getStatus() {
     return status;
   }
 
+  public long getAlertId() {
+    return id.orElse(DEFAULT_ID);
+  }
```

Comment:
Was wondering what your reasoning was on returning `DEFAULT_ID` for the alert id versus just the optional?

Copied and pasted from the PR description:
"Since an Alert object will only have an ID once it is queried from the Datastore, so it is likely that when creating Alert objects there is no ID. So, I made the ID field an OptionalLong so that the ID field can be empty when there is no ID to specify. Also, I added static factory methods to be used in place of constructors with reasoning explained below."

Reply:
My reasoning for returning a `long` instead of an `OptionalLong` is that I wanted to hide from the clients using the `Alert` class that i implemented the ID field as an optional. I don't want the clients having to worry about having to deal with an `OptionalLong` i.e. having to do `getAsLong()` and dealing with exception or `id.orElse(...)`. So, I designed the class so that the client can directly retrieve a `long` value and then use that `long` value directly to query an entity from the datastore.

However, the downside of directly returning a `long` instead of the optional, would be that the ID returned may not always be a valid ID that can be used to query an entity from the datastore. 

Even so, I decided to go with not returning an optional, as I assume (could be very bad to make this assumption) that the only time the client would ever want/need to `getAlertId()` it would be in the case where client is using an `Alert` object that is created from an entity (which definitely has a valid id field). In my opinion, the case where the client would try to retrieve an ID of an `Alert` object that is not created from an entity is very very rare, as if everything is done as designed, only the mock will be creating `Alert` object without an ID. 

I am not sure if this is the best decision given the pros/cons, so I would really appreciate if you can let me know what you think :)

Reply:
Thanks for the explanation. OK with the approach you've taken but please clarify that the DEFAULT_ID = 0 implies that the current Alert is invalid, so make this clear in the comment for now.

Also on a separate note, the assumption you made "getAlertId() it would be in the case where client is using an Alert object that is created from an entity (which definitely has a valid id field)" isn't entirely true since we can't control how other people and when other people use the classes/functions we create.